lib/src/model/product.dart

```diff
@@ -951,4 +969,75 @@ class Product extends JsonObject {
     }
     _nutriments = nutriments;
   }
+
+  /// Returns the best version of a product name.
+  ///
+  /// cf. openfoodfacts-server/lib/ProductOpener/Products.pm
+  String getBestProductName(final OpenFoodFactsLanguage language) {
+    String? tmp;
+    if ((tmp = productNameInLanguages?[language])?.isNotEmpty == true) {
+      return tmp!;
+    }
+    if ((tmp = productName)?.isNotEmpty == true) {
+      return tmp!;
+    }
+    if ((tmp = genericNameInLanguages?[language])?.isNotEmpty == true) {
+      return tmp!;
+    }
+    if ((tmp = genericName)?.isNotEmpty == true) {
+      return tmp!;
+    }
+    if ((tmp = abbreviatedNameInLanguages?[language])?.isNotEmpty == true) {
+      return tmp!;
+    }
+    if ((tmp = abbreviatedName)?.isNotEmpty == true) {
+      return tmp!;
+    }
```

Comment:
Can't we do a 

tmp ??= x;
tmp ??= y;

return tmp;

Reply:
@M123-dev Looks tempting but we check "not null and not empty", not just "not null", therefore your suggestion would potentially return empty strings.